ra/ra.go

```diff
@@ -1366,19 +1366,27 @@ func (ra *RegistrationAuthorityImpl) checkCertificatesPerNameLimit(ctx context.C
 }
 
 func (ra *RegistrationAuthorityImpl) checkCertificatesPerFQDNSetLimit(ctx context.Context, names []string, limit ratelimit.RateLimitPolicy, regID int64) error {
-	count, err := ra.SA.CountFQDNSets(ctx, &sapb.CountFQDNSetsRequest{
+	issuanceTimestampsInWindow, err := ra.SA.FQDNSetTimestampsForWindow(ctx, &sapb.CountFQDNSetsRequest{
 		Domains: names,
 		Window:  limit.Window.Duration.Nanoseconds(),
 	})
 	if err != nil {
 		return fmt.Errorf("checking duplicate certificate limit for %q: %s", names, err)
 	}
+
 	names = core.UniqueLowerNames(names)
 	threshold := limit.GetThreshold(strings.Join(names, ","), regID)
-	if count.Count >= threshold {
+	issuanceCount := int64(len(issuanceTimestampsInWindow.Timestamps))
+	if issuanceCount <= 0 {
```

Comment:
I believe this if-clause is unnecessary -- the previous structure of "if count >= threshold, return error; return nil" should still work.

Reply:
See conversation in https://github.com/letsencrypt/boulder/pull/6225/files#r929416918. If threshold is 0, and we get zero results, we would wind up indexing into the first element of Timestamps, which would panic.

Reply:
Ah right! But if threshold is zero, why are we querying the database at all? No matter the result from the db, we'll be returning a "threshold exceeded" rate limit error, and we should short-circuit all of the logic much earlier.

Could move the GetThreshold call to be the first line of this function, and bail out immediately if the result is zero.